sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/LocalDocumentStorage.java

```diff
@@ -267,56 +264,69 @@ void deletePendingOperation(PendingOperation pendingOperation) {
         return result;
     }
 
-    void updatePendingOperation(PendingOperation operation) {
-
-        /*
-            Update the document in cache (if expiration_time still valid otherwise, remove the document),
-            clear the pending_operation column, update etag, download_time and document columns
-         */
-        long now = Calendar.getInstance().getTimeInMillis();
-        if (operation.getExpirationTime() <= now) {
-            deletePendingOperation(operation);
-        } else {
-            mDatabaseManager.replace(getTableName(operation.getPartition()), getContentValues(operation, now), PARTITION_COLUMN_NAME, DOCUMENT_ID_COLUMN_NAME);
-        }
+    /**
+     * Deletes the specified document from the cache.
+     *
+     * @param operation Pending operation to delete.
+     */
+    void deletePendingOperation(PendingOperation operation) {
+        deleteOnline(operation.getTable(), operation.getPartition(), operation.getDocumentId());
     }
 
-    @VisibleForTesting
-    static String getTableName(String partition) {
-        return TABLE;
+    /**
+     * Validate partition name.
+     *
+     * @param partition name.
+     * @return true if the partition is supported, false otherwise.
+     */
+    static boolean isValidPartitionName(String partition) {
+        return READONLY.equals(partition) || USER.equals(partition);
     }
 
-    private static <T> ContentValues getContentValues(
+    private static ContentValues getContentValues(
             String partition,
             String documentId,
-            Document<T> document,
-            String etag,
+            String document,
+            String eTag,
             long expirationTime,
             long downloadTime,
             long operationTime,
             String pendingOperation) {
         ContentValues values = new ContentValues();
         values.put(PARTITION_COLUMN_NAME, partition);
         values.put(DOCUMENT_ID_COLUMN_NAME, documentId);
-        values.put(DOCUMENT_COLUMN_NAME, Utils.getGson().toJson(document));
-        values.put(ETAG_COLUMN_NAME, etag);
+        values.put(DOCUMENT_COLUMN_NAME, document);
+        values.put(ETAG_COLUMN_NAME, eTag);
         values.put(EXPIRATION_TIME_COLUMN_NAME, expirationTime);
         values.put(DOWNLOAD_TIME_COLUMN_NAME, downloadTime);
         values.put(OPERATION_TIME_COLUMN_NAME, operationTime);
         values.put(PENDING_OPERATION_COLUMN_NAME, pendingOperation);
         return values;
     }
 
+    void updatePendingOperation(PendingOperation operation) {
+
+        /*
+         * Update the document in cache (if expiration_time still valid otherwise, remove the document),
+         * clear the pending_operation column, update eTag, download_time and document columns.
+         */
+        long now = Calendar.getInstance().getTimeInMillis();
```

Comment:
`System.currentTimeMillis()` is slightly more efficient than `Calendar.getInstance().getTimeInMillis()` and we're already changing them in this PR. 

Why not use `System.currentTimeMillis()` here too?